nitric/resources/base.py

```diff
@@ -96,6 +96,11 @@ async def _register_policy_async(self, *args: str):
             raise exception_from_grpc_error(grpc_err)
 
     def _register_policy(self, *args: str):
-        loop = asyncio.get_event_loop()
-        loop.run_until_complete(self._register_policy_async(*args))
+        try:
+            loop = asyncio.get_event_loop()
+            loop.run_until_complete(self._register_policy_async(*args))
+        except RuntimeError:
+            # TODO: Check nitric runtime ENV variable
+            raise NitricResourceException("Nitric resource declared at runtime! Move resource declarations to top level of scripts")
```

Comment:
Nitric resources cannot be declared at runtime e.g. within the scope of a function. Move resource declarations to the top level of scripts so that they can be safely provisioned

Reply:
Suggesting a slightly more verbose description until we have docs that discuss this in greater detail.

Reply:
Your wording sounds much better. I think we should get the docs page going first and link this in now.